source/userguide/active_learning.rst

```diff
@@ -0,0 +1,269 @@
+
+
+.. _al_strategy_tuning:
+
+Strategy Tuning
+^^^^^^^^^^^^^^^
+The goal of the strategy tuning phase in the active learning pipeline is to determine the best tuning strategies and tuning level for your application. We will talk about the different possible tuning strategies and levels later in this section. 
+
+.. image:: /images/strategy_tuning.png
+    :align: center
+    :name: strategy_tuning_flow
+
+The existing training data in the app is partitioned into a `sampled` seed set and `unsampled` set according to the  ``train_seed_pct`` value mentioned in the config file. Data is evenly sampled at the ``train_seed_pct`` value across the domains to maintain class balance in the seed dataset. Note that the pipeline only uses the data from files that match the ``train_pattern`` regex in the config file in this step. The classifiers are trained on this sampled seed data (and evaluated on the existing test data, i.e. files matching ``test_pattern``).
+
+Next, the trained classifiers are used to generate a prediction for queries in the unsampled set. This prediction is output in the form of a class probability distribution. This distribution is passed through the sampling or ``tuning_strategies`` to obtain a ranked list of unsampled queries. Based on this ranked list, the top-k queries (k = ``batch_size``) are extracted from the unsampled set and added to the sampled set. Thereby increasing the size of the latter while reducing that of the former.
+
+The classifier models are now retrained with the expanded sampled set and evaluated against the same test set. This process is repeated until all the unsampled training data has been consumed by the sampled set and the final iteration of classifier training is done on this exhaustive sampled set. This tuning process is repeated for ``n_epochs`` (as defined in the config) to obtain average active learning performance.
+
+Information about unsampled queries selected at each iteration, and the performance of classifiers for each tuning strategy is stored iteratively in the ``output_folder`` directory. For every tuning command run, a new experiment folder is generated in this directory with the performance results and corresponding plots for a better visual understanding of the results.
+
+The following command can be used to run tuning using the settings defined in the application configuration:
+
+.. code-block:: console
+
+    mindmeld active_learning --tune --app-path '<PATH>/app_name/' --output_folder '<PATH>'
+
+Flags for application path and output folder are required and overwrite the default configuration settings for active learning. In addition to the aforementioned required flags, the following optional flags can be used - tuning_level, batch_size, n_epochs, train_seed_pct, and plot (default ``True``). These are described in detail in AL config section above.
```

Comment:
If they are required fields, they should not have defaults?

Reply:
Yeah the two don't? Rest are optional.

Reply:
"Flags for application path and output folder are required and overwrite the default configuration settings for active learning." --> "Flags for application path and output folder are required. In addition to the aforementioned required flags, the following optional flags can be used - tuning_level, batch_size, n_epochs, train_seed_pct, and plot (default ``True``). These are described in detail in AL config section above. These flags overwrite the default configuration settings for active learning."